pkg/config/kube/conversion.go

```diff
@@ -32,6 +32,17 @@ import (
 	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+var (
+	wellKnownPorts = map[int32]protocol.Instance{
+		25:    protocol.TCP,   // SMTP
+		53:    protocol.TCP,   // DNS. Default TCP if not specified.
+		3306:  protocol.MySQL, // MySQL
```

Comment:
This has nothing to do with enabling SQL. Its simply making sure that the protocol is not auto, and that we dont setup any sniffing logic as these are all server first protocols. Any sniffing will break these protocols (including tls inspector), even if we dont use mysql filter. Thats the reason people completely disable auth in Istio because it breaks mysql or mongo.

Reply:
my understanding is before if your port was named "mysql" you would get mysql. Now if your name is "mysql" OR your port is 3306 you get it.

Instead the behavior should just be if 3306 then don't enable sniffing , but don't set the protocol to tcp not mysql.


I don't think we should have magic ports at all but if we do at least just make it disable sniffing.

Rather than magic ports disabling sniffing, which will both be incomplete for many users and unexpected for others, I would think we just need to document that they can explicitly set the port to tcp/mysql as they wish to disable the sniffing? I guess that kind of goes against the 0 config goal we have with protocol sniffing though

Reply:
thats exactlu what is being done here: https://github.com/istio/istio/pull/15802/files#diff-6d0d125aefe784aaf86af0246ce9c91bR77

Reply:
it should probably be tcp though

Reply:
Change default to TCP.